File: semestr5/kursach/ARoom/ARoom.Services/Services/CategoryService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ARoom.Common.Model;
using ARoom.Data.IRepository;
using ARoom.Dto.Dtos;
using ARoom.Services.Contracts;
using AutoMapper;

namespace ARoom.Services.Services
{
    public class CategoryService : ICategoryService
    {
        private readonly ICategoryRepository _categoryRepository;
        private readonly IMapper mapper;

        public CategoryService(ICategoryRepository categoryRepository, IMapper mapper)
        {
            this._categoryRepository = categoryRepository;
            this.mapper = mapper;
        }

        public List<CategoryDto> GetCategories()
        {
            return this._categoryRepository.Query().Select(x => mapper.Map<CategoryDto>(x)).ToList();
        }

        public List<CategorySimpleDto> GetCategoriesSimple()
        {
            return this._categoryRepository.Query().Select(x => mapper.Map<CategorySimpleDto>(x)).ToList();
        }

        public void AddCategory(Category category)
        {
            this._categoryRepository.Add(category);
            this._categoryRepository.SaveChanges();
        }

        public void UpdateCategory(Category category)
        {
            this._categoryRepository.Update(category);
            this._categoryRepository.SaveChanges();
        }

        public void DeleteCategory(Category category)
        {
            this._categoryRepository.Remove(category);
            this._categoryRepository.SaveChanges();
        }
    }
}

File: semestr5/kursach/ARoom/ARoom/ClientApp/src/app/Categories/services/category.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

@Injectable({
    providedIn: 'root'
})
export class CategoryService {
    constructor(private http: HttpClient) {
        
    }

    getCategories(){
        return this.http.get('api/Category/GetCategories');
    }

    addCategory(cat: any){
        return this.http.post('api/category/AddCategory', cat);
    }
    updateCategory(cat: any){
        return this.http.post('api/category/UpdateCategory', cat)
    }
    deleteCategory(cat: any){
        return this.http.post('api/category/DeleteCategory', cat)
    }

    getCategoriesSimple(){
        return this.http.get('api/Category/GetCategoriesSimple');
    }
}